lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -195,12 +236,38 @@ def firewall_command(cmdlet_type)
 
           cmd
         end
+
+        # return the current firewall rule settings for icmp_type and protocol are correct
+        # @return [Boolean]
+        def is_set_properly?(icmp_type, protocol)
+          if icmp_type.to_s.empty?
+            return false # rubocop:disable Style/RedundantReturn
+
+          elsif icmp_type.is_a?(Integer)
+            return false unless protocol.start_with?("ICMP")
+            return false unless (0..255).include?(icmp_type)
+
+            true
+
+          elsif icmp_type.is_a?(String)
+            return false if !protocol.start_with?("ICMP") && icmp_type !~ /^\D+$/
+            return false if icmp_type.count("a-zA-Z") > 0 && icmp_type.count(":") > 1
+            return false unless (0..255).include?(icmp_type.to_i)
+
+            if protocol.start_with?("ICMP") && icmp_type.include?(":")
+              return icmp_type.split(":").all? { |type| (0..255).include?(type.to_i) }
+            end
+
+            # the following has to be disabled, as this works like a safeguard
+            true # rubocop:disable Style/RedundantReturn
```

Comment:
```suggestion
          case icmp_type
          when Integer
            protocol.start_with?("ICMP") && (0..255).include?(icmp_type)
          when String
            # string logic
          else
            false
          end

```

Like others have mentioned, this code could likely be simplified for more clarity.

I would have filled in the string logic but it was more complicated than I realized and felt like more work than it was worth trying to fix in a github comment box. :)